lib/Conversion/TritonToTritonGPU/TritonToTritonGPUPass.cpp

```diff
@@ -6,6 +6,7 @@
 #include "mlir/Dialect/Index/IR/IndexDialect.h"
 #include "mlir/Dialect/LLVMIR/LLVMDialect.h"
 #include "mlir/Dialect/LLVMIR/NVVMDialect.h"
+#include "mlir/Dialect/LLVMIR/ROCDLDialect.h"
```

Comment:
Are we using ROCLDialect here?

Reply:
We use ROCLDialect in any position where NVVM Dialect would be used.  I get this compilation error when the pass.td is converted to cpp. 

```  In file included from /dockerx/triton_rocm/lib/Conversion/TritonToTritonGPU/TritonToTritonGPUPass.cpp:21:
    /dockerx/triton_rocm/python/build/temp.linux-x86_64-cpython-38/include/triton/Conversion/Passes.h.inc: In member function ‘void ConvertTritonGPUToLLVMBase<DerivedT>::getDependentDialects(mlir::DialectRegistry&) const’:
    /dockerx/triton_rocm/python/build/temp.linux-x86_64-cpython-38/include/triton/Conversion/Passes.h.inc:348:25: error: ‘ROCDL’ is not a member of ‘mlir’
      348 |   registry.insert<mlir::ROCDL::ROCDLDialect>();
          |                         ^~~~~
    /dockerx/triton_rocm/python/build/temp.linux-x86_64-cpython-38/include/triton/Conversion/Passes.h.inc:348:12: error: parse error in template argument list
      348 |   registry.insert<mlir::ROCDL::ROCDLDialect>();```

Reply:
Hmm, but it's the optimizer? I am trying to understand why the TritonToTritonGPU pass depends on ROCM (or NVVM for that matter). Target-specific ops aren't added until the LLVM codegen

Reply:
TritonToTritonGPUPass has this inclusion. https://github.com/ROCmSoftwarePlatform/triton/blob/f171d6b20b78d6eb65fc2760c02842d90f17b9b6/lib/Conversion/TritonToTritonGPU/TritonToTritonGPUPass.cpp#L21-L22

Reply:
Do you want to do this refactor in this PR or in another PR?

Reply:
might as well doing here; seems like a small change

Reply:
I have a commit trying to refactor the conversion passes [here ](https://github.com/openai/triton/pull/1312/commits/18ca0948474ccdb395d0e833d5908f6405462f24) but I am getting this error.  
/dockerx/triton_rocm/lib/Conversion/TritonToTritonGPU/TritonToTritonGPUPass.cpp:22:10: fatal error: triton/Conversion/TritontoTritonGPU/Passes.h.inc: No such file or directory
       22 | #include "triton/Conversion/TritontoTritonGPU/Passes.h.inc"

Reply:
I think this refactor should be done in another pr by the code owner.

Reply:
@ptillet Done with the Conversion Pass refactor, the changes are in this [commit](https://github.com/openai/triton/pull/1312/commits/c785297b856ae58d1256523ddf70da9361901d87). It does increase the size of things from 25 to 34. Are you ok with that or do you want a new pr?